src/app/qgsstatisticalsummarydockwidget.cpp

```diff
@@ -159,6 +159,46 @@ void QgsStatisticalSummaryDockWidget::refreshStatistics()
   {
     mFieldType = fieldType( mFieldExpressionWidget->currentField() );
   }
+  else
+  {
+    QgsExpressionContext context;
```

Comment:
Would be nice to move this logic out to QgsExpressionUtils and avoid the duplicate code. Something like "determineResultType"